### PR TITLE
[BREAKING CHANGES] JSZip package upgrade

### PR DESCRIPTION
**Pull request by tiholic**
Upgraded JSZip to latest version 3.5.0 as per [upgrade guide](https://stuk.github.io/jszip/documentation/upgrade_guide.html)

This resulted in API changes as few methods required to be converted as async

- [x] Updated tests to use async/await wherever necessary
- [x] README updated
- [x] All tests passed

#### Pre release checklist
- [ ] Major Release version upgrade - add necessary release notes

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,10 +13,5 @@ checkouts
 .idea/vcs.xml
 .idea/workspace.xml
 atlassian-ide-plugin.xml
-.idea/dictionaries/vitalkond.xml
-.idea/inspectionProfiles/Project_Default.xml
-.idea/inspectionProfiles/profiles_settings.xml
-.idea/jsLibraryMappings.xml
-.idea/jsLinters/jshint.xml
-.idea/libraries/sass_stdlib.xml
+.idea
```
Comment (TrySound) [inline .gitignore:16]: Perhaps idea files above can be dropped as well.

**Comment by kant2002** [comment]
1. This require us to go to version 2.0
2. I would try to no create churn with breaking changes/version changes too much. While 2.0 would be backed, can we keep documentation for 1.0 intact and have 2.0 examples in readme somewhere in separate section. Maybe as separate file.
3. Also I agree in principle with this changes and that it is desirable, but can we have current version of JSZip package with mocked definitions like 
```
declare module 'jszip' {
  var x: any;
  export = x;
}
```

that way it give us time to think how make less disruptive changes. Maybe we can take inspiration from https://stackoverflow.com/questions/16586640/node-js-async-to-sync and provide backward compatible API + async alternatives.

**Comment by tiholic** [comment]
@kant2002 let me know what would you like to do with this PR. If you want to have a v2, then have it as a separate release? or maintain it under separate branch for now and do a release later!

**Comment by kant2002** [comment]
@tiholic I want you to show how can we have cake and eat it too. I believe we can achieve your goals without major changes in the API and everything will just works. I will keep it for reference, but eventually likely it would not be accepted.

**Comment by kant2002** [comment]
I made a fork of old JSzip for now and apply security fixes. So this is in its current form not needed. If you have idea how to expose async API so I can gradually migrate without disrupting users, I would appreciate it

**Pull request closed without merging**